### DISCUSSION TITLE
Updating from SimNIBS version 3.2.3 to 4.1.0 - TMS coil orientation

### DISCUSSION THREAD

**Discussion by eburkevt** [simnibs/simnibs#418] [6. Getting started] [answered]
Our lab has a been using code (bash scripts, afni, fsl, matlab) that uses version 3.2.3 of SimNIBS to find the optimum coil handle orientation for TMS stimulation. These scripts were developed at UPenn (2020) and the methods were described here [https://pubmed.ncbi.nlm.nih.gov/33005039/](url)

I am in the process of updating the code to version 4.1.0 of SimNIBS. However, the outputs of SimNIBS have changed and certain parts of our code need updating to match the different outputs.

One significant issue for us is that the normE nifti file that was output by v3.2.3 (For example, subject001_TMS_1-0001_Medtronic_MCF_B65_vn_normE.nii.gz) has been replaced with a 'magnE' nifti file (For example, subject001_TMS_1-0001_Medtronic_MCF_B65_vn_magnE.nii.gz) 

... and it looks as though, for magnE., only the gray matter voxels have E-field magnitude values. For normE, it appears there were E-field magnitude values for all tissue, including bone/skull/scalp and white matter as well as gray matter.

Here is the same subject showing (left) magnE (4.1.) output, and (right) normE (3.2.3) output ... and the red blob is the region (voxels), based on functional MRI data, from which we compute the average E-field strength for a particular coil orientation. You can see that on the left (magnE) only the gray matter ribbon has nonzero E values, while the right (normE) has E values for all brain tissue, including scalp/skull/bone.

![image](https://github.com/user-attachments/assets/33172418-e140-490a-bcc1-87914ce8db80)

Is this correct, or are there settings to add back e-field values for non-gray matter brain tissue (make magnE match normE)?

**Comment by TorgeW** [comment, marked as answer]
```
S = sim_struct.SESSION()
...
S.tissues_in_niftis = [1,2,3]  # Results in the niftis will be masked 
                               # to only show WM (1), GM (2), CSF(3)
                               # (standard: only GM)
                               # To get fields everywhere: 
                               #    S.tissues_in_niftis = 'all'
```

You can find an example using this here:
https://github.com/simnibs/simnibs/blob/master/simnibs/examples/simulations/tDCS_advanced.py

A list of all tissues can be found here:
https://github.com/simnibs/simnibs/blob/master/simnibs/utils/mesh_element_properties.py

**Comment by eburkevt** [comment]
Thank you for your help!
